keras_hub/src/models/task.py

```diff
@@ -325,21 +328,20 @@ def add_layer(layer, info):
                 )
 
             preprocessor = self.preprocessor
-            tokenizer = getattr(preprocessor, "tokenizer", None)
-            if tokenizer:
-                info = "Vocab size: "
-                info += highlight_number(tokenizer.vocabulary_size())
-                add_layer(tokenizer, info)
-            image_converter = getattr(preprocessor, "image_converter", None)
-            if image_converter:
-                info = "Image size: "
-                info += highlight_shape(image_converter.image_size())
-                add_layer(image_converter, info)
-            audio_converter = getattr(preprocessor, "audio_converter", None)
-            if audio_converter:
-                info = "Audio shape: "
-                info += highlight_shape(audio_converter.audio_shape())
-                add_layer(audio_converter, info)
+            if preprocessor and isinstance(preprocessor, keras.Layer):
```

Comment:
can you please explain why you have made these changes?

Reply:
This is needed for a nested preprocessor. I have added a comment for this change:

```python
            # Since the preprocessor might be nested with multiple `Tokenizer`,
            # `ImageConverter`, `AudioConverter` and even other `Preprocessor`
            # instances, we should recursively iterate through them.
```